BEP100.md

```diff
@@ -0,0 +1,154 @@
+
+```solidity
+interface INativeAssetBridge {
+
+  function deposit() external payable;
+
+  function withdraw(bytes[] validatorSetSignatures, bytes transactionReceipt) external;
+}
+```
+
+Native asset bridge supports next user flows:
++ Deposit (BAS -> BSC). When a user calls a deposit function it locks his native tokens in the smart contract and emits events. To be able to mint peg tokens in BSC chain, the user generates a proof that contains information about transaction receipt (including emitted events) with merkle patricia trie proof. This proof should be uploaded into the BSC chain where a `BASValidatorHub` smart contract can verify this proof and validators signatures from the BAS chain.
++ Withdrawal (BSC -> BAS). Withdrawal of the funds is opposite to deposit flow. The difference here is that the user must burn his pegged tokens in the BSC chain to use this proof to withdraw tokens from native asset bridge smart contracts. Validators in the BAS network must verify the correctness of this operation and are responsible for preventing double spend attacks.
+
+BAS developers might specify fees for inter-chain operations and cross chain operations between BAS and BSC chains. Such policy should be specified in the BAS smart contract and fees or fines related mechanisms shouldn’t be presented in BSC smart contracts or block verification functions.
+
+### 5.1  BAS Validator Hub
+
+By default BSC brings only native asset cross chain functionality to the BAS applications. To reach this, each BAS application must be registered in the BSC smart contract. This smart contract assigns chain id to the BAS application and specifies BHVF (block header verification function). This function is used to verify the block header, but w/o state transitions. BHVF should be written in Solidity and be able to verify block header, the correctness of the chain for N blocks and check signatures. Since default BAS implementation is BSC-based then default chain implementation is supported by BAS validator hub, we define such function to allow other developers that don’t want to rely on EVM or parlia consensus engine to write their own verification function. Of course developers might publish malicious functions or a function with a vulnerability that allows them to skip block verification and they should provide an audition of this code to be trusted by a community.
+
+```solidity
+interface IBlockHeaderVerificationFunction {
+    function verifyBlockHeader(bytes rawBlockHeader, address[] existingValidatorSet) external pure returns (address[] newValidatorSet);
+    
+    function verifyCrossChainPacket(bytes rawBlockHeader, bytes receiptsProof, bytes rawReceipt) external pure returns (
+        uint256 transferredAmount,
+        address recipientAddress
+    );
+}
+
+interface BASValidatorHub {
+  function registerBAS(string projectName, address[] initialValidatorSet, IBlockHeaderVerificationFunction bhvf) external;
```

Comment:
I think `BASValidatorHub` do not manage the `ValidatorSet` for BAS, it is managed in different `bhvf` contract or other contract thyself,  since different BAS may have different validator structure.

Reply:
Yes, validators are managed by BAS and they should be stored in the storage after validator transition is confirmed. This interfaces is removed from doc since it's useless.